### PR TITLE
Port changes to FloatingScreen from upstream.

### PR DESCRIPTION
**Pull request by FrozenAlex**
Needs a bit of review but I tried to be as close to the original as possible

### CODE REVIEW

File: src/BSML/FloatingScreen/FloatingScreenHandle.cpp [hunk 1]
```diff
@@ -1,31 +1,143 @@
...
+        Vector3 targetPosition = _grabbingController->_viewAnchorTransform->get_transform()->TransformPoint(_grabPos);
+        Quaternion targetRotation = _grabbingController->_viewAnchorTransform->get_transform()->get_rotation() * _grabRot;
+
+        _floatingScreen->get_transform()->SetPositionAndRotation(
+            Vector3::Lerp(_floatingScreen->get_transform()->get_position(), targetPosition, 10.0f * Time::get_unscaledDeltaTime()),
+            Quaternion::Lerp(_floatingScreen->get_transform()->get_rotation(), targetRotation, 5.0f * Time::get_unscaledDeltaTime())
+        );
+    }
+    
+    void FloatingScreenHandle::OnDestroy() {
```
Comment (FrozenAlex) [inline src/BSML/FloatingScreen/FloatingScreenHandle.cpp:131]: I'm not sure if we even need to set these to nullpointers

File: src/BSML/FloatingScreen/FloatingScreenHandle.cpp [hunk 2]
```diff
@@ -1,31 +1,143 @@
...
+        _grabbingController = nullptr;
+        _floatingScreen = nullptr;
+    }
```
Comment (FrozenAlex) [inline src/BSML/FloatingScreen/FloatingScreenHandle.cpp:134]: I'm not sure if we even need to clear those on destroy

File: src/BSML/FloatingScreen/FloatingScreen.cpp [hunk 3]
```diff
@@ -164,7 +151,6 @@ namespace BSML {
 
     void FloatingScreen::OnDestroy() {
```
Comment (FrozenAlex) [inline src/BSML/FloatingScreen/FloatingScreen.cpp:152]: OnDestroy is probably not needed here anymore. I don't see any uses for it

File: src/BSML/FloatingScreen/FloatingScreenHandle.cpp (hunk 2 above)
Comment (RedBrumbler) [reply src/BSML/FloatingScreen/FloatingScreenHandle.cpp:134]: probably not needed but it doesn't matter all that much

**Pull request merged**